### DISCUSSION TITLE
Question about fabric event

### DISCUSSION THREAD

**Discussion by bh4rtp2023** [hyperledger/fabric#4329] [Q&A] [answered]
Does fabric generate only one event only when the invocation is correctly endorsed?
I want to send error event but failed.

**Comment by bestbeforetoday** [comment, marked as answer]
If you are talking about chaincode events...

The short answer is that the client API only emits chaincode events for successfully committed / validated transactions.

The intent of chaincode eventing is to drive business processes in response to updates to the ledger. Only successful transactions update the ledger, and so only chaincode events for successful transactions are reported. See description in the [Fabric Full Stack Development workshop](https://github.com/hyperledger/fabric-samples/blob/main/full-stack-asset-transfer-guide/docs/ApplicationDev/05-ChaincodeEvents.md).

The long answer is that, if a transaction gets as far as being committed in a block (in other words: endorsed, and then successfully submitted to the orderer), even transactions that fail validation (perhaps because of an MVCC read conflict, or failure to meet endorsement requirements) are still recorded on the blockchain along with their chaincode events. Using block eventing you can iterate over the transactions in a block and dig down to the chaincode event for failed transactions.

**Comment by bh4rtp2023** [comment]
@bestbeforetoday Thank you for your answer. 
I can get error message from [gateway example](https://github.com/hyperledger/fabric-samples/blob/9ee0aac2ae3ac57e27c3210751922b735b0ea556/asset-transfer-basic/application-gateway-go/assetTransfer.go#L228).
And I can get the transaction id and the endorsers after `contract.Submit()` when an error occurs, but how to get those results when the invocation is successful?
